psycopg/utils.c

```diff
@@ -50,8 +50,13 @@ psycopg_escape_string(connectionObject *conn, const char *from, Py_ssize_t len,
     Py_ssize_t ql;
     int eq = (conn && (conn->equote)) ? 1 : 0;
 
-    if (len == 0)
+    if (len == 0) {
         len = strlen(from);
+    } else if (strcspn(from, "\0") != len) {
```

Comment:
I think strcspn is appropriate. strcspn(s, reject) spans until one of the characters in reject is found, not until all of the characters are found. i.e. it is not doing string comparison, it is meant for the case where i'm using it for. I did doubt if I shouldn't use strlen(from), but I decided not to, because with strcspn it is more obvious that we're looking for a NUL terminator inside a string.

strchr returns a pointer, which can also be NULL. So it would be more verbose to implement it like that. i.e. it would read something like, which in my opinion is less readable.
``
char *p = strchr(from, '\0');

if (p != NULL && p - from != len) {
 // exception
} 
``